### PR TITLE
Feature/data resource

### PR DESCRIPTION
**Pull request by daiLlew**
### What

Added functionality to create and update data resources.

### How to review

You should be able to create and update data resources.

### Who can review

@CarlHembrough @YetAnotherMatt @NeilMadden @githubmo @fenallen

### CODE REVIEW

**Review by YetAnotherMatt — CHANGES_REQUESTED** [review]
As per Dave's comment, this requires changes

File: src/main/java/uk/co/onsdigital/discovery/dao/DataResourceDAOImpl.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    @Override
+    public void update(DataResource dataResource) throws DataResourceException {
+        try {
+            NamedParam.ListBuilder builder = new NamedParam.ListBuilder()
+                    .addParam(DATA_RESOURCE_FIELD, dataResource.getDataResourceID())
+                    .addParam(TITLE_FIELD, dataResource.getTitle())
+                    .addParam(METADATA_FIELD, dataResource.getMetadata());
+            jdbcTemplate.update(UPDATE_DATA_RESOURCE_SQL, namedParameterFactory.create(builder));
+        } catch (DataAccessException ex) {
+            ex.printStackTrace();
```
Comment (YetAnotherMatt) [inline src/main/java/uk/co/onsdigital/discovery/dao/DataResourceDAOImpl.java:68]: Can we replace all the ex.printStackTrace with logger.error(...)?

File: src/main/java/uk/co/onsdigital/discovery/api/DataResourceAPI.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    @Autowired
+    private DataResourceDAO dataResourceDAO;
+
+    /**
+     * Create a new {@link DataResource}
+     */
+    @RequestMapping(value = "/dataResource", method = RequestMethod.POST)
+    public ResponseEntity<CreatedResponse> createDataResource(@Valid @RequestBody DataResource dataResource,
+                                                              BindingResult bindingResult)
+            throws DataResourceException {
```
Comment (YetAnotherMatt) [inline src/main/java/uk/co/onsdigital/discovery/api/DataResourceAPI.java]: formatting seems to have gone a bit weird here. That's how picky I'm being ;)

**Review by YetAnotherMatt — APPROVED** [review]
A couple of comments that shouldn't prevent merging

File: src/main/java/uk/co/onsdigital/discovery/dao/DataResourceDAOImpl.java (hunk 1 above)
Comment (daiLlew) [reply src/main/java/uk/co/onsdigital/discovery/dao/DataResourceDAOImpl.java:68]: " Its alpha & throw away "

**Review by YetAnotherMatt — APPROVED** [review]

**Pull request merged**